src/App.tsx

```diff
@@ -1,10 +1,48 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { Status } from './types/Status';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
 
-const USER_ID = 0;
+const USER_ID = 11092;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [todosFilterBy, setTodosFilterBy] = useState<Status>(Status.ALL);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch((error) => {
+        setIsError(true);
+        throw error;
+      });
+  }, []);
+
+  const preparedTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (todosFilterBy) {
+        case Status.ALL:
+          return todos;
+
+        case Status.ACTIVE:
+          return !todo.completed;
+
+        case Status.COMPLETED:
+          return todo.completed;
+
+        default:
+          throw new Error('Unable to add a todo');
+      }
```

Comment:
move filtering inside switch and return filtered todos.
Status.All redundant. just return todos in default case.
```suggestion
      switch (todosFilterBy) {
        case Status.ACTIVE:
          return todos.filter(todo => !todo.completed);

        case Status.COMPLETED:
          return todos.filter(todo => todo.completed);

        default:
           return todos;
      }
```